components/cam/src/dynamics/se/stepon.F90

```diff
@@ -443,7 +432,12 @@ subroutine stepon_run2(phys_state, phys_tend, dyn_in, dyn_out )
       call outfld('DYN_Q'     ,dyn_in%elem(ie)%state%Q(:,:,:,1)       ,npsq,ie)
       call outfld('DYN_U'     ,dyn_in%elem(ie)%state%V(:,:,1,:,tl_f)  ,npsq,ie)
       call outfld('DYN_V'     ,dyn_in%elem(ie)%state%V(:,:,2,:,tl_f)  ,npsq,ie)
-      call outfld('DYN_PS'    ,dyn_in%elem(ie)%state%ps_v(:,:,tl_f)   ,npsq,ie)
+
+      ! at this point in the code we are on floating lagrangian levels
+      ! need to compute ps for output
+      p_i(:,:,nlevp) = hvcoord%hyai(1)*hvcoord%ps0 + &
```

Comment:
Seems like the calculation of p_i below is redundant with this here, so it's probably a good idea to remove the one lower down

Reply:
Isn't the p_i calculation below inside of a compiler flag, so not always guaranteed?

Reply:
Right, but this part is not within an ifdef, so whether or not that section gets triggered by running with a non-theta-l configuration the values of p_i will already be calculated, so no need to calculate them twice.